### ISSUE TITLE
replaceRegexpAll has different behavior in 21 vs 20, and doesn't match the documentation

### ISSUE THREAD

**Issue by cpcloud** [ClickHouse/ClickHouse#32777]
**Describe the unexpected behaviour**

I expect

```sql
select replaceRegexpAll('Hello, World!', '^', 'here: ')
```

to produce 

```
here: Hello, World!
```

in v20, this is the case, and in v21 the result is:

```
here: ello, World!
```

**How to reproduce**

Run the above query in clickhouse v20, then in v21

**Comment by KochetovNicolai** [comment]
Indeed:
```
Connected to ClickHouse server version 21.13.1 revision 54453.

:) select replaceRegexpAll('Hello, World!', '^', 'here: ')

SELECT replaceRegexpAll('Hello, World!', '^', 'here: ')

Query id: 31ca9523-377b-44f4-acf0-604c188cfac5

┌─replaceRegexpAll('Hello, World!', '^', 'here: ')─┐
│ here: ello, World!                               │
└──────────────────────────────────────────────────┘
```

**Comment by sulayliu** [comment]
connected ClickHouse client version 21.11.6.7 (official build).
```
select replaceRegexpAll('Hello, World!', '^', 'here: ');
```
SELECT replaceRegexpAll('Hello, World!', '^', 'here: ')

Query id: 1b716d63-5624-43f8-850f-4a0723dcd44b

┌─replaceRegexpAll('Hello, World!', '^', 'here: ')─┐
│ here: ello, World!                               │
└──────────────────────────────────────────────────┘

**Comment by den-crane** [comment]
@sulayliu it was fixed after 21.11.6.7 has been released.

It was fixed only in 21.12. It will be never fixed in 21.11 https://github.com/ClickHouse/ClickHouse/pull/32945

```
Connected to ClickHouse server version 21.12.3 revision 54452.

SELECT replaceRegexpAll('Hello, World!', '^', 'here: ')

┌─replaceRegexpAll('Hello, World!', '^', 'here: ')─┐
│ here: Hello, World!                              │
└──────────────────────────────────────────────────┘
```
